### PR TITLE
Primary set of CLI stubs for Smart Class Parameter

### PR DESCRIPTION
**Pull request by jyejare**
This is primary set of CLI stubs for feature Smart Class Parameter.
And more to come in next PR.

CLI Usage:
```
# hammer -u <user> -p <passwd> sc-param --help
Usage:
    hammer sc-param [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    subcommand
 [ARG] ...                     subcommand arguments

Subcommands:
 add-override-value            Create an override value for a specific smart variable
 info                          Show a smart class parameter
 list                          List all smart class parameters
 remove-override-value         Delete an override value for a specific smart variable
 update                        Update a smart class parameter

```

### CODE REVIEW

**Comment by sthirugn** [comment]
All these are cli tests but they mention `checkbox`, `submit` and other UI related terminologies.  Is that intentional?
- Does this feature have a cli counterpart? If so can you please rephrase these tests accordingly?

**Comment by jyejare** [comment]
@sthirugn : submit is intentional but not checkbox i mean to say check there.

**Comment by jyejare** [comment]
@sthirugn , Please re-review the updated PR. Thanks !!

File: tests/foreman/cli/test_classparameters.py [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+    def test_positive_validate_puppet_default_value(self):
+        """Validation doesn't works on puppet default value.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Set puppet default value to 'Use Puppet Default'.
+        3.  Validate this value under section 'Optional Input Validator'.
```
Comment (sthirugn) [inline tests/foreman/cli/test_classparameters.py]: is this applicable to cli test?

File: tests/foreman/cli/test_classparameters.py [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+    @tier3
+    def test_negative_validate_matcher_value_with_regex(self):
+        """Error raised for matcher value not matching with regex.
+
+        @feature: Class Parameters - Optional Validation
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Create a matcher with value that doesn't matches
```
Comment (sthirugn) [inline tests/foreman/cli/test_classparameters.py]: s/matches/match

File: tests/foreman/cli/test_classparameters.py [hunk 3]
```diff
@@ -0,0 +1,827 @@
...
+    def test_positive_override_check(self):
+        """Override the Default Parameter value.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Set the new valid Default Value.
+        3.  Submit the changes.
```
Comment (sthirugn) [inline tests/foreman/cli/test_classparameters.py]: `Submit the changes` is implied in cli commands, so this is not required.

File: tests/foreman/cli/test_classparameters.py [hunk 4]
```diff
@@ -0,0 +1,827 @@
...
+    @tier3
+    def test_positive_override_check(self):
+        """Override the Default Parameter value.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Override the parameter.
+        2.  Set the new valid Default Value.
```
Comment (sthirugn) [inline tests/foreman/cli/test_classparameters.py]: Steps 1 and 2 mean the same thing here right?

File: tests/foreman/cli/test_classparameters.py (hunk 3 above)
Comment (sthirugn) [inline tests/foreman/cli/test_classparameters.py]: @jyejare for this cli test, I dont think `@steps` section is required at all.  The test name `Override the default parameter value` is sufficient enough to explain the test.  What is your opinion?

Comment (sthirugn) [inline tests/foreman/cli/test_classparameters.py]: if you agree with my comment above, can you also please fix other tests?

File: tests/foreman/cli/test_classparameters.py (hunk 4 above)
Comment (jyejare) [inline tests/foreman/cli/test_classparameters.py]: Both are different options in CLI command. The intention is to set override to True before setting the value.

File: tests/foreman/cli/test_classparameters.py (hunk 3 above)
Comment (jyejare) [inline tests/foreman/cli/test_classparameters.py]: See my answer on your questions 'Steps 1 and 2 mean the same thing here right?'.

File: tests/foreman/cli/test_classparameters.py (hunk 1 above)
Comment (jyejare) [inline tests/foreman/cli/test_classparameters.py]: Yes, As asset says it should not work with puppet default value. The intention is to 'NOT' raise any error for puppet default value.

**Comment by sthirugn** [comment]
TY @jyejare for answering my questions.

**Comment by sthirugn** [comment]
Although the steps explained here is not so intuitive for a cli test (it is for a ui test), but this is not a blocker and can be adjusted/updated when the tests are implemented.  I am okay for now.

**Comment by sthirugn** [comment]
ACK

**Comment by rplevka** [comment]
ACK

**Pull request merged**